flocker/node/_docker.py

```diff
@@ -508,6 +530,12 @@ def _remove():
                     break
 
             try:
+                # The ``docker.Client.stop`` method puts the container into
+                # a stopping state, but the container cannot be removed
+                # until the container has actually stopped running.
+                while self._blocking_container_runs(container_name):
```

Comment:
I'd disagree that it is not covered by tests. If you take this code out, you get more test failures. Remember, we're supposed to be testing external behaviour in the tests, not internal details.

There is actually no deterministic behaviour to be tested, due to FLOC-1729. After a `DockerClient.add`, we cannot know what state to expect, and after a `DockerClient.remove`, there is no container to be tested.

@wallrj had some ideas about using a fake Docker Server, but I'm not sure it's worth the effort at this stage.